Fix passing params to status server

kapliars opened the pull request:
Pass configuration section "status" as status thin server parameters. Fixes bug with Component register method which expects port, username and password as options, but these values are not read from config file and not passed. No it takes section "status" from yaml config and passes it to Component.register(options)

dieu committed:
Refactoring configuration status server

derekcollison commented:
The port and user/pass credentials are generated randomly so to decrease the attack surface. These endpoints can be discovered by sending a discovery message on the message bus..

kapliars commented:
Derek, they are still generated if not specified in config. 

But when you need you may specify them. Access to NATS is no more secure: afaik we keep mbus credentials plain in config file - what's the use to have more secure status server? And random password / port really complicates monitoring system configuring. Particularly after restart of CF admin has to reconfigure iptables and monitoring system - is not it crazy? System configuration when status server is closed with firewall from outer world - is common and simple solution to achieve desired security level.

What could be more common and reliable to add security than to move configuration files out of codebase to common place with limited access like /etc/ on linux systems.

By the way I could not manage to get this info from discovery message: as far as I see NATS does not support fanout at the moment and request is always sent to the last topic subscriber - which means I always get discovery response only from gateway component. The only way to fetch it I found is to listen to vcap.announce messages, still it requires additional listener proccess.

derekcollison commented:
Sending a single discovery message will have all vcap components respond with information that gives access to the http endpoints. After a restart, sending one simple message will get all the information you need. You need to have the listener wait for all the responses.

kapliars commented:
cool, you're right: discover works for now, yet it works with NATS.request, with NATS.publish with reply_to I still get only one response... 

If support for providing these options to Thin server is already here, what's the reason not to allow user to it? I would like to use it, I know how to deal with static passwords (by the way there's no way to change password without restart - that is security problem), why can not I do?

If I dont want this, I remove (literally speaking do not add) entry from config file and I'm totally safe with random password.

derekcollison commented:
NATS.publish with a reply_to that is subscribed to without auto_unsubscribe should also work.

Let us think about your request for static config, sounds reasonable.

kapliars committed:
Add comments to status server config

- Make status config optional, use random values if not provided in config
- Comment out status server config so it is not specified by default.
- Add description to config

kapliars commented:
I have updated pull request - now by default config is commented out and if it is not provided random values are generated

pbozeman commented:
pulled.  thanks
